### PR TITLE
Charity Maps Screen

### PR DESCRIPTION
**Pull request by PreyanshK**
This PR does the following:

- Used Google Maps API to display the fridges and the user's location on a map
- Live Location Tracking of the user's device every 10s (so the marker **follows** the user)
- Used the Android's Geocoder API to convert location addresses to [Lat, Long] coordinates (which will be useful when the user adds a new fridge, and we want to display it on the map)
- Fridge Repository connection to fetch the fridge locations from Charity Collection from Firestore
- Updated the schema for Charity Collection to include GeoPoint for the Fridge

Demo:

https://github.com/adshayanB/ECE452/assets/36520183/09b0f42d-e8e6-4b75-935a-88f8b1a919a9

Firestore:
image

### CODE REVIEW

File: app/src/main/java/com/example/farmeraid/data/CharityRepository.kt [hunk 1]
```diff
@@ -57,17 +60,19 @@ class CharityRepository (
...
-            val itemsMap = items as MutableMap<String, Int>
-            val fridgeName = name as String
-            val charityLocation = location as String
+        return if (items != null && name !=null && location!=null && coordinates != null) {
+            val itemsMap = items
+            val fridgeName = name
+            val charityLocation = location
             ResponseModel.FAResponseWithData.Success(
                 FridgeModel.Fridge(
+                    id = id,
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/data/CharityRepository.kt:75]: I realized my changes hasnt come here yet, but I will rebase my pr to support the geo stuff

**Review by adshayanB — APPROVED** [review]
You are a beast

**Review by RishanR — APPROVED** [review]
Looking good

**Review by SoumilDharaskar — APPROVED** [review]
LGTM! Great work!

**Review by SathursanS — APPROVED** [review]
LGTM!

**Pull request merged**